walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -419,6 +419,36 @@ public void optionalAttendeeAvailable() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void timesOutOfOrder() {
+    // Based on everyAttendeeIsConsidered, add events to the event collection 
+    // out of order by start time. It should return the same output 
+    //
+    // Events  :       |--A--|     |--B--|
+    //                       |--C--|
+    // Day     : |-----------------------------|
+    // Options : |--1--|                 |--2--|
+ 
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 2", TimeRange.fromStartDuration(TIME_0900AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_B)),
+        new Event("Event 3", TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_C)),
+        new Event("Event 1", TimeRange.fromStartDuration(TIME_0800AM, DURATION_30_MINUTES),
+            Arrays.asList(PERSON_A)));
+ 
+    MeetingRequest request =
+        new MeetingRequest(Arrays.asList(PERSON_A, PERSON_B, PERSON_C), DURATION_30_MINUTES);
+ 
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0800AM, false),
+            TimeRange.fromStartEnd(TIME_0930AM, TimeRange.END_OF_DAY, true));
+ 
+    Assert.assertEquals(expected, actual);
+  }
+
```

Comment:
We need more test cases to uncover the bug. For starters, try this test case (given to the test in order from top to bottom)

```
     |---A--------------|
                                                  |----B---|
             |-----C---|
                                             |---------D--------|
```
where time is left-to-right.

Reply:
There's already a test that looks like this:
// Events  :        |----A----|
//                         |--B--|
// Day       : |---------------------|
// Options : |--1--|          |--2--|

I think that would have similar results to the A/C test, but if not let me know and I'll throw a test like that in (or just revise that test to have the times in a different order?)

I'll add a test for the B/D scenario though

Reply:
(that comment didn't quite transfer the B was supposed to be nested within a)
But I just realized you wanted 4 different events in the same test case, given in the order A, B, C , D where C is nested in A and B is nested in D, right?

Reply:
Right.

I think another test that might fail right now is simply: 2 events, in reverse chronological order, with enough space in between for the to-be-scheduled event. Unless I'm misunderstanding your code.

Reply:
Okay so the test I wrote did pass. Let me know if that wasn't what you meant though

Reply:
Yeah it turns out I was wrong this whole time. Your code works great. Sorry about that.